python/lsst/pipe/tasks/diff_matched_tract_catalog.py

```diff
@@ -0,0 +1,306 @@
+        columns_all = [self.coord_format.column_target_coord1, self.coord_format.column_target_coord2,
+                       self.column_target_extended]
+        if self.coord_format.coords_ref_to_convert is not None:
+            columns_all.extend(self.coord_format.coords_ref_to_convert.values())
+        for columns_list in (
+            (
+                self.columns_target_coord_err,
+                self.columns_target_select_false,
+                self.columns_target_select_true,
+                self.columns_target_copy,
+            ),
+            (x.columns_in_target for x in self.columns_flux.values()),
+        ):
+            for columns in columns_list:
+                columns_all.extend(columns)
+        return set(columns_all)
+
+    columns_flux = pexConfig.ConfigDictField(
+        keytype=str,
+        itemtype=MatchedCatalogFluxesConfig,
+        doc="Configs for flux columns for each band",
+    )
+    columns_ref_copy = pexConfig.ListField(
+        dtype=str,
+        default=set(),
+        doc='Reference table columns to copy to copy into cat_matched',
+    )
+    columns_target_coord_err = pexConfig.ListField(
+        dtype=str,
+        listCheck=lambda x: (len(x) == 2) and (x[0] != x[1]),
+        doc='Target table coordinate columns with standard errors (sigma)',
+    )
+    columns_target_copy = pexConfig.ListField(
+        dtype=str,
+        default=('patch',),
+        doc='Target table columns to copy to copy into cat_matched',
+    )
+    columns_target_select_true = pexConfig.ListField(
+        dtype=str,
+        default=('detect_isPrimary',),
+        doc='Target table columns to require to be True for selecting sources',
+    )
+    columns_target_select_false = pexConfig.ListField(
+        dtype=str,
+        default=('merge_peak_sky',),
+        doc='Target table columns to require to be False for selecting sources',
+    )
+    coord_format = pexConfig.ConfigField(
+        dtype=ConvertCatalogCoordinatesConfig,
+        doc="Configuration for coordinate conversion",
+    )
+
+
+class DiffMatchedTractCatalogTask(pipeBase.PipelineTask):
+    """Load subsets of matched catalogs and output a merged catalog of matched sources.
+    """
+    ConfigClass = DiffMatchedTractCatalogConfig
+    _DefaultName = "DiffMatchedTractCatalog"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        skymap = inputs.pop("skymap")
+
+        outputs = self.run(
+            catalog_ref=inputs['cat_ref'].get(parameters={'columns': self.config.columns_in_ref}),
+            catalog_target=inputs['cat_target'].get(parameters={'columns': self.config.columns_in_target}),
+            catalog_match_ref=inputs['cat_match_ref'].get(
+                parameters={'columns': ['match_candidate', 'match_row']},
+            ),
+            catalog_match_target=inputs['cat_match_target'].get(
+                # parameters={'columns': ['match_candidate']},
```

Comment:
Should this be commented out?

Reply:
It's from DM-33221, so I suppose I can just delete it for now.